tests/kubernetes/test_pod_generator.py

```diff
@@ -658,8 +658,8 @@ def test_deserialize_model_file(self):
     def test_pod_name_confirm_to_max_length(self, _, pod_id):
         name = PodGenerator.make_unique_pod_id(pod_id)
         assert len(name) <= 253
-        parts = name.split(".")
-        if len(pod_id) <= 63:
+        parts = name.split("-")
+        if len(pod_id) <= 63 - 33:
             assert len(parts[0]) == len(pod_id)
         else:
             assert len(parts[0]) <= 63
```

Comment:
```suggestion
        parts = name.split("-")
        assert parts[0] == pod_id[:30]
```

Also the following line I can't suggest, line 666, no longer makes sense imo. We should check it against a fixed length, I think this is right:

```
assert len(parts[1]) == 32
```

Reply:
This makes much more sense. thanks!